great_expectations_provider/common/gx_context_actions.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
+
+if TYPE_CHECKING:
+    from great_expectations import ExpectationSuite
+    from great_expectations.core.batch_definition import BatchDefinition
+    from great_expectations.core.expectation_validation_result import (
+        ExpectationSuiteValidationResult,
+    )
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.expectations import Expectation
+
+
+def run_validation_definition(
+    task_id: str,
+    expect: Expectation | ExpectationSuite,
+    batch_definition: BatchDefinition,
+    result_format: Literal["BOOLEAN_ONLY", "BASIC", "SUMMARY", "COMPLETE"] | None,
+    batch_parameters: dict,
+    gx_context: AbstractDataContext,
+) -> ExpectationSuiteValidationResult:
+    """Given a BatchDefinition and an Expectation or ExpectationSuite, ensure a
+    ValidationDefinition and run it."""
+    import great_expectations as gx
+
+    if isinstance(expect, gx.expectations.Expectation):
+        suite = gx.ExpectationSuite(name=task_id, expectations=[expect])
+    else:
+        suite = expect
+    validation_definition = gx_context.validation_definitions.add_or_update(
+        validation=gx.ValidationDefinition(
+            name=task_id,
+            suite=suite,
+            data=batch_definition,
+        ),
+    )
+    if result_format:
```

Comment:
Complete nit but branching like this always feels like a smell. Core has a `DEFAULT_RESULT_FORMAT` constant that is used as the default. Might make this stuff a bit more clean to just use that.

Reply:
yeah, i considered that. The problem is we can't reference GX imports top level in airflow, and a function signature is top level, so we're not able to use the core `DEFAULT_RESULT_FORMAT`. Another option would be to update the `ValidationDefinition.run` signature to allow `None`, and have GX handle it, or to just set the kwarg default manually here, like `result_format="BOOLEAN"`. But against those options, this one seemed the most straight forward, but happy to discuss further, or go with another option, if you feel it would be better.